net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+\t\t<string>${calendarserverdir}/Library/CalendarServer/Config/calendarserver.plist</string>|" \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver/org.macports.calendarserver.plist
+
+    # org.macports.calendarserver_proxy
+    reinplace \
+        "s|<string>${prefix}/sbin/nginx</string>$|&\\
+\t<string>-g</string>\\
+\t<string>daemon off;</string>\\
+\t<string>-p</string>\\
+\t<string>${calendarserverdir}/Library/CalendarServer/logs</string>\\
+\t<string>-c</string>\\
+\t<string>${calendarserverdir}/Library/CalendarServer/etc/nginx.conf</string>|" \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver_proxy/org.macports.calendarserver_proxy.plist
+    reinplace \
+        "s|<string>${prefix}/var/run/calendarserver_proxy.pid</string>$|<string>${calendarserverdir}/Library/CalendarServer/run/nginx.pid</string>|" \
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver_proxy/org.macports.calendarserver_proxy.plist
+
+    # TLS PFS
+    if ![file exists \
+        ${calendarserverdir}/Library/CalendarServer/etc/dhparam.pem] {
+        # create a shorter, faster DH parameter file for the default installation
+        system -W ${calendarserverdir}/Library/CalendarServer/etc \
+            "openssl dhparam -out dhparam.pem 2048"
+    }
+
+    fs-traverse f_or_d \
+        ${calendarserverdir} {
+        if { [file type ${f_or_d}] != "link" } {
+            file attributes ${f_or_d} \
+                 -owner ${calendarserverUser} -group _calendar
+        }
+    }
+}
+
+notes "Apple Calendar and Contacts Server is a standards-compliant
+server implementing the CalDAV and CardDAV protocols. Full
+deployment requires a working mail server, DNS configuration on both the
+LAN and the internet, including SPF and DKIM records, trusted TLS
+certificates, port forwarding, possibly a mail relay, and more.
+
+Users must reconfigure this installation for their own system, network,
+and security model specifics by editing all necessary files and checking
+file permissions. A subset of these settings are visible in the files:
+
+        port contents calendar-contacts-server
+        port file calendar-contacts-server
+
+These are the locations and network settings for the default configuration:
+
+    Calender and Contacts Server:
+        ${prefix}/var/calendarserver/Library/CalendarServer/Config/calendarserver.plist
+
+    nginx Reverse Proxy:
+        ${prefix}/var/calendarserver/Library/CalendarServer/etc/nginx.conf
+        ports: 8008, 8443, 8800, 8843
+
+A working Calendar and Contacts Server will allow local account authentication
+at these web pages (ports 8008 and 8800 are unencrypted):
+
+        http://${fullhost}:8008
+        https://${fullhost}:8443
+        http://${fullhost}:8800
+        https://${fullhost}:8843
+
+It is necessary to set the active PostgreSQL binaries to postgresql96 using
+the command:
+
+        sudo port select --set postgresql postgresql96
```

Comment:
This is very bad. Couldn't the sources be patched to reference the correct binary?

Reply:
I don’t think it’s too bad. The source use the `psql` and associated binaries, like every other package that uses PostgreSQL.

This is the way MacPorts provides to create the `psql` executable.

PostgreSQL comes with a ton of executables and symlinks to other executables. I don’t think it’s very robust to modify code to point to version-specific binaries.

I’m reluctant to do this unless someone files a ticket saying this actually breaks something.

Reply:
I used `reinplsace as you suggested. This is gone now.`